src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -4863,6 +4939,401 @@ resource pipeline_RerunETL 'Microsoft.DataFactory/factories/pipelines@2018-06-01
+                  value: '@activity(\'Load Schema Mappings\').output.firstRow.translator'
+                  type: 'Expression'
+                }
+              }
+              inputs: [
+                {
+                  referenceName: 'resourcegraph'
+                  type: 'DatasetReference'
+                  parameters: {}
+                }
+              ]
+              outputs: [
+                {
+                  referenceName: 'ingestion'
+                  type: 'DatasetReference'
+                  parameters: {
+                    blobPath: {
+                      value: '@concat(variables(\'blobBasePath\'), item().type, \'.parquet\')'
+                      type: 'Expression'
+                    }
+                  }
+                }
+              ]
+            }
+            { name: 'Catch Query Failure'
+              type: 'IfCondition'
+              dependsOn: [
+                {
+                  activity: 'Execute Query'
+                  dependencyConditions: [
+                    'Failed'
+                  ]
+                }
+              ]
+              userProperties: []
+              typeProperties: {
+                expression: {
+                  value: '@contains(activity(\'Execute Query\').output.errors[0].Message, \'Sequence contains no elements\')'
+                  type: 'Expression'
+                }
+                ifFalseActivities: [
+                  {
+                    name: 'Set Query Error Counter'
+                    type: 'SetVariable'
+                    dependsOn: []
+                    policy: {
+                      secureOutput: false
+                      secureInput: false
+                    }
+                    userProperties: []
+                    typeProperties: {
+                      variableName: 'pipelineFailed'
+                      value: {
+                        value: '@bool(true)'
+                        type: 'Expression'
+                      }
+                    }
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      }
+      { // Overall Exports Check
+        name: 'Overall Exports Check'
+        type: 'IfCondition'
+        dependsOn: [
+          {
+            activity: 'Iterate Queries'
+            dependencyConditions: [
+              'Completed'
+            ]
+          }
+        ]
+        userProperties: []
+        typeProperties: {
+          expression: {
+            value: '@equals(variables(\'pipelineFailed\'), true)'
+            type: 'Expression'
+          }
+          ifTrueActivities: [
+            {
+              name: 'Fail Pipeline'
+              type: 'Fail'
+              dependsOn: []
+              userProperties: []
+              typeProperties: {
+                message: {
+                  value: 'Pipeline failed'
+                  type: 'Expression'
+                }
+                errorCode: 'ARGQueriesFailed'
```

Comment:
We should make the prefix dynamic and pull from the JSON file.

Reply:
Can you be more specific? This activity is very generic, being there to raise a pipeline error only when at least one of the queries raised an error that is not attributable to a result with empty rows. This is needed to ensure the pipeline finishes successfully in case no cost recommendations were found.

Reply:
I meanwhile understood what you meant. However, given the nature of each source type (ARG, ADX, etc.), error handling will probably be specific to the source type. We can discuss this further offline.